src/metabase/api/alert.clj

```diff
@@ -32,10 +32,13 @@
   [archived user_id]
   {archived [:maybe ms/BooleanString]
    user_id  [:maybe ms/PositiveInt]}
-  (as-> (pulse/retrieve-alerts {:archived? (Boolean/parseBoolean archived)
-                                :user-id   user_id}) <>
-    (filter mi/can-read? <>)
-    (t2/hydrate <> :can_write)))
+  (let [user-id (if api/*is-superuser?*
+                  user_id
+                  api/*current-user-id*)]
```

Comment:
Could you update the docstring to mention that the `user_id` param is ignored if the caller isn't an admin?